src/api.c

```diff
@@ -1321,6 +1321,60 @@ static void cgroup_free_cg_mount_table(void)
 	memset(&cg_cgroup_v2_empty_mount_paths, 0, sizeof(cg_cgroup_v2_empty_mount_paths));
 }
 
+/*
+ * Parses the mount options of the given mount point and checks for the
+ * option in the list of mount options.
+ * @mnt: Mount point name to search for mount points.
+ * @mnt_option: Mount option to be searched.
+ *
+ * Returns 0, in case of success and ECGOTHER on failure.
+ */
+static int check_mount_point_opt(char *mnt, char *mnt_option)
+{
+	struct mntent *ent, *temp_ent = NULL;
+	char mntent_buffer[4 * FILENAME_MAX];
+	char *mntopt = NULL;
+	FILE *proc_mount;
+	int ret = 0;
+
+	proc_mount = setmntent(mnt, "r");
+	if (!proc_mount) {
+		cgroup_err("cannot open %s: %s\n", mnt, strerror(errno));
+		last_errno = errno;
+		ret = ECGOTHER;
+		goto err;
+	}
+
+	temp_ent = (struct mntent *) malloc(sizeof(struct mntent));
+	if (!temp_ent) {
+		last_errno = errno;
+		ret = ECGOTHER;
+		goto err;
+	}
+
+	ent = getmntent_r(proc_mount, temp_ent,	mntent_buffer, sizeof(mntent_buffer));
+	if (!ent) {
+		last_errno = errno;
+		ret = ECGOTHER;
+		goto err;
+	}
+
+	mntopt = hasmntopt(ent, mnt_option);
+	if (!mntopt) {
+		cgroup_dbg("%s, not found in the list of ", mnt_option);
+		cgroup_cont("mount options of %s\n", mnt);
+		ret = ECGOTHER;
+	}
+
+err:
+	if (proc_mount)
+		fclose(proc_mount);
```

Comment:
Per `setmntent`'s man page, should we be using `endmntent()` rather than `fclose()`?

```
       The setmntent() function opens the filesystem description file filename and returns a file pointer which can
       be  used  by  getmntent().  The argument type is the type of access required and can take the same values as
       the mode argument of fopen(3).   The  returned  stream  should  be  closed  using  endmntent()  rather  than
       fclose(3).
```

Reply:
My bad, impatient me did not read the `setmntent()` man page thoroughly. Sure will change it.